polyphemus/lib/metis/mvir1_waiver.rb

```diff
@@ -25,13 +25,125 @@ def pool_search_string(pool_name)
     "#{pool_name}"
   end
 
-  def restrict_patient_data(patient_name)
-    @metis_client.rename_folders(
+  def diff_worker_for(release_folder, restricted_folder) end
+
+  def empty_for(from_bucket, to_bucket, folder_name)
+    Etna::Clients::Metis::WalkMetisDiffWorkflow.new(
+      left_walker: Etna::Clients::Metis::WalkMetisWorkflow.new(
+        metis_client: @metis_client,
+        project_name: @project_name,
+        bucket_name: from_bucket,
+        root_dir: folder_name,
+      ),
+      right_walker: Etna::Clients::Metis::WalkMetisWorkflow.new(
+        metis_client: @metis_client,
+        project_name: @project_name,
+        bucket_name: to_bucket,
+        root_dir: folder_name,
+      ),
+    ).each do |diff|
+      type, left, right = diff
+
+      case type
+      when :equal
+        try_empty_file(left)
+      when :left_unique
+        try_copy(left, to_bucket)
+        try_empty_file(left)
+      when :right_unique
+        # leave it be, maybe an artifact of concurrent behavior
+      else
+        try_remove(right)
+        try_copy(left, to_bucket)
+        try_empty_file(left)
+      end
+    end
+  end
+
+  # Try to atomically empty the contents of the file without it disappearing.
+  def try_empty_file(file)
+    return unless file.is_a?(Etna::Clients::Metis::File)
+
+    uploader = Etna::Clients::Metis::MetisUploadWorkflow.new(
+      metis_client: @metis_client,
       project_name: @project_name,
-      source_bucket: @release_bucket_name,
-      source_folders: find_folders_in_bucket(@release_bucket_name, patient_name_search_string(patient_name)),
-      dest_bucket: @restrict_bucket_name,
+      bucket_name: file.bucket_name,
+      metis_uid: '7af8e4be837744ce791c6239d00afefd',
+    )
+
+    uploader.do_upload(
+      Etna::Clients::Metis::MetisUploadWorkflow::StreamingIOUpload.new(
+        readable_io: StringIO.new
+      ),
+      file.file_path
+    )
+  end
+
+  def try_copy(file, to_bucket)
+    if file.is_a?(Etna::Clients::Metis::Folder)
+      req = Etna::Clients::Metis::CreateFolderRequest.new(
+        project_name: @project_name,
+        bucket_name: to_bucket,
+        folder_path: file.folder_path
+      )
+      @metis_client.create_folder(req) unless @metis_client.folder_exists?(req)
+      return
+    end
+
+    downloader = Etna::Clients::Metis::MetisDownloadWorkflow.new(
+      metis_client: @metis_client,
+      project_name: @project_name,
+      bucket_name: file.bucket_name
     )
+
+    io = StringIO.new
+    downloader.do_download(io, file)
+    io.rewind
+
+    uploader = Etna::Clients::Metis::MetisUploadWorkflow.new(
+      metis_client: @metis_client,
+      project_name: @project_name,
+      bucket_name: to_bucket,
+      metis_uid: '7af8e4be837744ce791c6239d00afefd',
+    )
+
+    uploader.do_upload(
```

Comment:
Some of these files can be quite large ... does this mean they will all be streamed down / up from Polyphemus, instead of using the Metis `copy` API? Will that be efficient if there are a significant number of large files?